src/ITensorChainRules/ITensorChainRules.jl

```diff
@@ -321,13 +321,17 @@ function ChainRulesCore.rrule(::typeof(apply), x1::Vector{ITensor}, x2::MPS; kwa
   return y, apply_pullback
 end
 
-function ChainRulesCore.rrule(::typeof(apply), x1::Vector{ITensor}, x2::MPO; apply_dag::Bool = true, kwargs...)
+function ChainRulesCore.rrule(
+  ::typeof(apply), x1::Vector{ITensor}, x2::MPO; apply_dag::Bool=true, kwargs...
```

Comment:
Putting `apply_dag::Bool=true` wouldn't guarantee that it is true, it is just setting a default. In fact it may be worse and actually change the default behavior of the forward pass, I'm not exactly sure what the interplay is between the keyword arguments defined in the original function vs. the `rrule` of the function.

We should add a check:
```julia
!apply_dag && errror("rrule for apply MPO not yet defined for `apply_dag=false`")
```
or something like that.

Reply:
This function handles the case with `apply_dag=false` as well.

Reply:
Sure, just need to grab its value inside the function. Will make this change